v0/destinations/ProfitWell/transform.js

```diff
@@ -0,0 +1,182 @@
+const get = require("get-value");
+const logger = require("../../../logger");
+const { EventType } = require("../../../constants");
+const { getSubscriptionHistory } = require("./nethandler");
+const {
+  CustomError,
+  getDestinationExternalID,
+  defaultRequestConfig,
+  defaultPostRequestConfig,
+  defaultPutRequestConfig,
+  removeUndefinedAndNullValues,
+  constructPayload,
+  getSuccessRespEvents,
+  getErrorRespEvents,
+  getFieldValueFromMessage
+} = require("../../util");
+const { ConfigCategory, mappingConfig, baseEndpoint } = require("./config");
+
+const identifyResponseBuilder = async (message, { Config }) => {
+  const response = defaultRequestConfig();
+
+  const user_id = getDestinationExternalID(message, "profitwellUserId");
+  const user_alias = getFieldValueFromMessage(message, "userId");
+
+  if (!user_id && !user_alias) {
+    throw new CustomError("UserId not found", 404);
+  }
+
+  let subscriptionId = getDestinationExternalID(
+    message,
+    "profitwellSubscriptionId"
+  );
+  let subscriptionAlias =
+    get(message, "traits.subscriptionAlias") ||
+    get(message, "context.traits.subscriptionAlias");
+
+  const targetUrl = `${baseEndpoint}/v2/users/${user_id || user_alias}`;
+  const res = await getSubscriptionHistory(targetUrl, {
+    headers: {
+      Authorization: Config.privateApiKey
```

Comment:
we should put a check to see if `Config.privateApiKey` exists or not. better to put that check inside `process`. If invalid, we should throw a CustomeError() with appropriate message to drop the event